### PR TITLE
[KYUUBI #5358] Bump iceberg 1.4.0

### PR DESCRIPTION
**Pull request by wForget**
### _Why are the changes needed?_

https://github.com/apache/iceberg/releases/tag/apache-iceberg-1.4.0

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_

No

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2190,6 +2190,7 @@
                 <spark.version>3.1.3</spark.version>
                 <spark.binary.version>3.1</spark.binary.version>
                 <delta.version>1.0.1</delta.version>
+                <iceberg.version>1.3.1</iceberg.version>
```
Comment (wForget) [inline pom.xml:2193]: Iceberg 1.4.0 removes support for spark 3.1

Comment (pan3793) [reply pom.xml:2193]: let's document it in the comment

"Iceberg removed the support for Spark 3.1 in apache/iceberg#xxxx since 1.4.0"

Comment (wForget) [reply pom.xml:2193]: Iceberg removed the support for Spark 3.1 in apache/iceberg#8661 since 1.4.0

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Merged to master/1.8

**Pull request closed without merging**